Source/WebSocket.swift

```diff
@@ -185,11 +185,18 @@ public class WebSocket : NSObject, NSStreamDelegate {
         guard isConnected else { return }
         dequeueWrite(str.dataUsingEncoding(NSUTF8StringEncoding)!, code: .TextFrame)
     }
-    
-    ///write binary data to the websocket. This sends it as a binary frame.
-    public func writeData(data: NSData) {
+
+    /**
+     Write binary data to the websocket. This sends it as a binary frame.
+     
+     If you supply a non-nil completion block, I will perform it when the write completes.
+
+     - parameter data:       The data to write.
+     - parameter completion: The (optional) completion handler.
+     */
+    public func writeData(data: NSData, completion: (() -> ())?) {
```

Comment:
Can we set `completion` to default to nil? 

`writeCompletion: (() -> ())? = nil`

That way we can avoid can breaking changes for current versions that upgrade.